File: easy/problems/0387-first-unique-character-in-a-string/solution.ts
function firstUniqChar(s: string): number {
    const count = new Map<string, number>();

    for (const char of s) {
        count.set(char, (count.get(char) || 0) + 1);
    }

    for (let i = 0; i < s.length; i++) {
        if (count.get(s[i]) === 1) {
            return i;
        }
    }

    return -1;
}

File: easy/problems/0387-first-unique-character-in-a-string/solution.py
def firstUniqChar(s):
    """
    Find the first unique character in a string.

    Args:
        s: str - input string

    Returns:
        int - index of first unique character, -1 if none
    """
    from collections import Counter

    char_count = Counter(s)

    for i, char in enumerate(s):
        if char_count[char] == 1:
            return i

    return -1
